src/EncompassRest.Tests/LoanTests.cs

```diff
@@ -357,5 +357,171 @@ public void Loan_ExtensionData_Serialization()
+            Assert.IsNull(loan.Fields["HMDA.X32"]);
+            Assert.AreEqual(@"{""hmda"":{""income"":null}}", loan.ToJson());
+        }
+
+        [TestMethod]
+        public void Loan_FieldsCustomString()
+        {
+            var loan = new Loan();
+            var value = "ABC";
+            loan.Fields["CX.NAME"] = value;
+            Assert.AreEqual(value, (string)loan.Fields["CX.NAME"]);
+            Assert.AreEqual($@"{{""customFields"":[{{""fieldName"":""CX.NAME"",""stringValue"":""{value}""}}]}}", loan.ToJson());
+            loan.Fields["CX.NAME"] = null;
+            Assert.IsNull(loan.Fields["CX.NAME"]);
+            Assert.AreEqual(@"{""customFields"":[{""fieldName"":""CX.NAME"",""stringValue"":null}]}", loan.ToJson());
+        }
+
+        [TestMethod]
+        public void Loan_FieldsCustomDate()
+        {
+            var loan = new Loan();
+            var value = DateTime.Now;
+            loan.Fields["CX.NOW"] = value;
+            Assert.AreEqual(value, (DateTime?)loan.Fields["CX.NOW"]);
+            Assert.AreEqual($@"{{""customFields"":[{{""dateValue"":{value.ToJson()},""fieldName"":""CX.NOW""}}]}}", loan.ToJson());
+            loan.Fields["CX.NOW"] = null;
+            Assert.IsNull(loan.Fields["CX.NOW"]);
+            Assert.AreEqual(@"{""customFields"":[{""dateValue"":null,""fieldName"":""CX.NOW""}]}", loan.ToJson());
+        }
+
+        [TestMethod]
+        public void Loan_FieldsCustomNumeric()
+        {
+            var loan = new Loan();
+            var value = 1234.56M;
+            loan.Fields["CX.NUMBER"] = value;
+            Assert.AreEqual(value, (decimal?)loan.Fields["CX.NUMBER"]);
+            Assert.AreEqual($@"{{""customFields"":[{{""fieldName"":""CX.NUMBER"",""numericValue"":{value}}}]}}", loan.ToJson());
+            var integerValue = 98765;
+            loan.Fields["CX.NUMBER"] = integerValue;
+            Assert.AreEqual(integerValue, (decimal?)loan.Fields["CX.NUMBER"]);
+            Assert.AreEqual($@"{{""customFields"":[{{""fieldName"":""CX.NUMBER"",""numericValue"":{integerValue}.0}}]}}", loan.ToJson());
+            loan.Fields["CX.NUMBER"] = null;
+            Assert.IsNull(loan.Fields["CX.NUMBER"]);
+            Assert.AreEqual(@"{""customFields"":[{""fieldName"":""CX.NUMBER"",""numericValue"":null}]}", loan.ToJson());
+        }
+
+        [TestMethod]
+        public void Loan_FieldsBP()
+        {
+            var loan = new Loan();
+            var value = "Joe";
+            loan.Fields["4000#2"] = value;
+            Assert.AreEqual(value, (string)loan.Fields["4000#2"]);
+            Assert.AreEqual($@"{{""applications"":[{{""applicationIndex"":1,""borrower"":{{""firstName"":""{value}""}}}}]}}", loan.ToJson());
+            loan.Fields["4000#2"] = null;
+            Assert.IsNull(loan.Fields["4000#2"]);
+            Assert.AreEqual($@"{{""applications"":[{{""applicationIndex"":1,""borrower"":{{""firstName"":null}}}}]}}", loan.ToJson());
+        }
     }
```

Comment:
Add Loan_FieldsInvalidField and LoanFieldsInvalidCustomField
Both should throw ArgumentException